external/storm-hdfs-oci/src/main/java/org/apache/storm/container/oci/LocalOrHdfsImageTagToManifestPlugin.java

```diff
@@ -0,0 +1,294 @@
+            if (localImageToHash != null && !localImageToHash.equals(localImageToHashCache)) {
+                localImageToHashCache = localImageToHash;
+                LOG.info("Reloaded local image tag to hash cache");
+                ret = true;
+            }
+        }
+
+        try (BufferedReader hdfsBr = getHdfsImageToHashReader()) {
+            Map<String, String> hdfsImageToHash = readImageToHashFile(hdfsBr);
+            if (hdfsImageToHash != null && !hdfsImageToHash.equals(hdfsImageToHashCache)) {
+                hdfsImageToHashCache = hdfsImageToHash;
+                LOG.info("Reloaded hdfs image tag to hash cache");
+                ret = true;
+            }
+        }
+        return ret;
+    }
+
+    private BufferedReader getLocalImageToHashReader() throws IOException {
+        if (localImageTagToHashFile == null) {
+            LOG.debug("Did not load local image to hash file, file is null");
+            return null;
+        }
+
+        File imageTagToHashFile = new File(localImageTagToHashFile);
+        if (!imageTagToHashFile.exists()) {
+            LOG.warn("Did not load local image to hash file, file doesn't exist");
+            return null;
+        }
+
+        long newLocalModTime = imageTagToHashFile.lastModified();
+        if (newLocalModTime == localModTime) {
+            LOG.debug("Did not load local image to hash file, file is unmodified");
+            return null;
+        }
+        localModTime = newLocalModTime;
+
+        return new BufferedReader(new FileReader(imageTagToHashFile));
+    }
+
+    private BufferedReader getHdfsImageToHashReader() throws IOException {
+        if (hdfsImageToHashFile == null) {
+            LOG.debug("Did not load hdfs image to hash file, file is null");
+            return null;
+        }
+
+        Path imageToHash = new Path(hdfsImageToHashFile);
+        FileSystem fs = imageToHash.getFileSystem(new Configuration());
+        if (!fs.exists(imageToHash)) {
+            String message = "Could not load hdfs image to hash file, " + hdfsImageToHashFile + " doesn't exist";
+            LOG.error(message);
+            throw new IOException(message);
+        }
+
+        long newHdfsModTime = fs.getFileStatus(imageToHash).getModificationTime();
+        if (newHdfsModTime == hdfsModTime) {
+            LOG.debug("Did not load hdfs image to hash file, file is unmodified");
+            return null;
+        }
+        hdfsModTime = newHdfsModTime;
+
+        return new BufferedReader(new InputStreamReader(fs.open(imageToHash)));
+    }
+
+    // You may specify multiple tags per hash all on the same line.
+    // Comments are allowed using #. Anything after this character will not
+    // be read
+    // Example file:
+    // foo/bar:current,fizz/gig:latest:123456789
+    // #this/line:wont,be:parsed:2378590895
+    //
+    // This will map both foo/bar:current and fizz/gig:latest to 123456789
```

Comment:
Is this supposed to be part of javadoc on parsing logic?

Reply:
will convert this to javadoc